src/psyclone/tests/parse/kernel_test.py

```diff
@@ -133,15 +180,59 @@ def test_getkernelfilepath_caseinsensitive2(tmpdir):
     assert "tmp" in result
     assert "test_mod.f90" in result
 
-# function get_kernel_metadata
+
+# function get_kernel_interface
+def test_get_kernel_interface_match():
+    ''' Tests that something is returned when searching for an interface'''
+    module_parse_tree = parse(CODE_INTERFACE)
+    meta1, meta2 = get_kernel_interface(module_parse_tree)
+    assert meta1 is not None
+    assert meta2 is not None
+
+
+def test_get_kernel_interface_no_match():
+    ''' Tests for none when searching there is no interface'''
+    module_parse_tree = parse(CODE)
+    meta1, meta2 = get_kernel_interface(module_parse_tree)
+    assert meta1 is None
+    assert meta2 is None
 
 
-def test_get_kernel_metadata_no_match():
+def test_get_kernel_interface_match_caseinsensitive():
+    ''' Tests that the interface name is case insensitive'''
+    module_parse_tree = parse(CODE_INTERFACE.replace("test_code", "TeST_CoDe"))
+    meta1, meta2 = get_kernel_interface(module_parse_tree)
+    assert meta1 == "test_code"
+    assert meta2 is not None
+
+
+def test_get_kernel_interface_match_correct():
+    ''' Tests that something is returned when searching for an interface'''
+    module_parse_tree = parse(CODE_INTERFACE.replace(
+        "module procedure sub_code",
+        "module procedure sub_code, more_code"))
+    meta1, meta2 = get_kernel_interface(module_parse_tree)
+    assert meta1 == "test_code"
+    assert meta2[0] == "sub_code"
+    assert meta2[1] == "more_code"
+
+
+def test_get_kernel_interface_double_interface():
+    ''' Tests that parse error occurs for two interfaces.'''
```

Comment:
"...occurs when the parse tree contains two interfaces." (Subject to agreement on the precise rules we are going to specify for LFRic kernels.)

Reply:
Done.